src/drive/ducks/upload/index.js

```diff
@@ -125,3 +172,22 @@ export const getUploadQueue = state => state[SLUG].queue
 export const getProcessed = state =>
   getUploadQueue(state).filter(f => f.status !== PENDING)
 export const getSuccessful = state => getLoaded(getUploadQueue(state))
+
+// DOM helpers
+const extractFilesEntries = items => {
+  let results = []
+  for (let i = 0; i < items.length; i += 1) {
+    const item = items[i]
+    if (item.webkitGetAsEntry != null && item.webkitGetAsEntry()) {
+      const entry = item.webkitGetAsEntry()
+      results.push({
+        file: item.getAsFile(),
+        isDirectory: entry.isDirectory === true,
+        entry
+      })
+    } else {
+      results.push({ file: item, isDirectory: false, entry: null })
+    }
+  }
+  return results
```

Comment:
I think we could use a `map()` call here, to be more in line with the rest of the code base?

Reply:
Not sure it's possible to use a map, because these APIs use some kind of weird iterators. Maybe with an `Array.from`...